Kerpecheva/scheduler new features (#769)
telerik/maui-docs: controls/scheduler/built-in-dialogs.md

**1.** Create a sample view with a `RadScheduler` instance and a few buttons:

<snippet id='scheduler-dialogs-methods-xaml' />

**2.** Add the buttons' clicked event handlers in the code-behind:

<snippet id='scheduler-dialogs-methods-eventhandlers' />

## Events

The Scheduler exposes the following events for its built-in dialogs:

* `DialogOpening`&mdash;Raised when a dialog is about to be opened. The `DialogOpening` event handler receives two parameters:
    * The sender argument, which is of type `object`, but can be cast to the `RadScheduler` type.
    * A `SchedulerDialogOpeningEventArgs` object with the following properties:
       * `DialogType`&mdash;Provides the dialog type.
       * `Cancel`&mdash;If `True`, cancels the dialog opening.
* `DialogClosing`&mdash;Raised when a dialog is about to be closed. The `DialogClosing` event handler receives two parameters:
    * The sender argument, which is of type `object`, but can be cast to the `RadScheduler` type.
    * A `SchedulerDialogClosingEventArgs` object with the following properties:
       * `DialogType`&mdash;Provides the dialog type.
       * `Cancel`&mdash;If `True`, cancels the dialog closing.

>        * `DialogResult (bool?)`&mdash;Indicates how the dialog was closed. `null` when closed with the **Close** button. `False` when closed with the **Cancel** button. `True` when <BE SPECIFIC>.

Reviewer: Check the end of the line :) Either provide explicit details or go with "`True` in all other cases."

Check a quick example on how to use the `DialogOpening` and `DialogClosing` events:

**1.** Add the `RadScheduler` definition:

<snippet id='scheduler-dialogs-events-xaml' />

**2.** Add the event handlers:

<snippet id='scheduler-dialogs-events' />

## Customization

You can style the visual appearance of each dialog by creating the corresponding `Style` with `TargetType` set to `SchedulerDialog`. Here is a list of the available `Style` properties:

> * `EditAppointmentDialogStyle`&mdash;Defines the `Style` that will be applied on the Edit Appointment dialog.

Reviewer: I probably missed this the first time around (sorry for that). Go with "Defines the `Style` for the " for all bullets :)

> ---

Reviewer: The page doesn't have an inline ToC ("In this article", in the sidebar). That's due to the heading levels. Consider switching to H2s for Methods, Events, and Customization.

title: Built-in Dialogs
page_title: .NET MAUI Scheduler Documentation - Built-in Dialogs
description: Learn more about the Telerik .NET MAUI Scheduler built-in dialogs for creating, editing and deleting appointments.
position: 9

> slug: scheduler-built-in dialogs

Author: Thank you, fixed

---

# .NET MAUI Scheduler Built-in Dialogs

Telerik .NET MAUI Scheduler control exposes built-in dialogs for creation and modification of appointments, so you can provide users with the ability to directly schedule their meetings.

> Check below the available dialogs for managing appointments.

Reviewer: Consider re-working the list into H4s. Right now the screenshots don't nest that well visually. 

For example:

```
#### Dialog Name

Description from the bullet.

![Screenshots](...)
```

Here's how that would look like, if you decide to go that way.

![image](https://github.com/telerik/maui-docs/assets/3539221/0fc4c041-8912-47a5-8afc-af4a68561eb9)

Author: Agree

> * **PreviewAppointment Dialog**&mdash;Used only on Android and iOS. When the end user taps on an existing appointment, the following dialog appears, giving the option to edit or delete the appointment:

Author: Switching all to their names (with spaces)

    * A `SchedulerDialogOpeningEventArgs` object which you can use to get the dialog type through `DialogType` property as well as cancel the opening by setting its `Cancel` property to `True`.
* `DialogClosing`&mdash;occurs when a dialog is about to be closed. The `DialogClosing` event handler receives two parameters:
    * The sender argument, which is of type `object`, but can be cast to the `RadScheduler` type.
    * A `SchedulerDialogClosingEventArgs` object which you can use to get the dialog type through `DialogType` property as well as cancel the opening by setting its `Cancel` property to `True`. In addition, the `SchedulerDialogClosingEventArgs` provides a `DialogResult`(`bool?`) property which gives information on how the dialog is closed: `DialogResult` is null when the dialog is closed by the Close button of the dialog, `False` when Cancel option is pressed, otherwise the value is `True`.

Check a quick example on how the `DialogOpening` and `DialogClosing` events can be used:

**1.** Add the `RadScheduler` definition:

<snippet id='scheduler-dialogs-events-xaml' />

**2.** Add the event handlers:

<snippet id='scheduler-dialogs-events' />

> ### Customization

Reviewer: Based on what users can achieve with the corresponding dialog styles, consider what the best name for the sections is.
Go with Customization, if users can use these styles to change the labels or further configure the UI elements of the built-in dialog (say, place them vertically, instead of horizontally, etc.).
Go with Styling, if it's just about colors, borders, font types, etc.

**It might also make sense to extract this into the Styling section and provide some brief examples of actual styling.**

> You can customize the visual appearance of each dialog by creating the corresponding style with TargetType set to `SchedulerDialog`. Here is a list of the available Style properties:

Reviewer: ```suggestion
You can style the visual appearance of each dialog by creating the corresponding `Style` with `TargetType` set to `SchedulerDialog`. Here is a list of the available `Style` properties:
```

> * `EditAppointmentDialogStyle`&mdash;Defines the Style that will be applied on the edit appointment dialog control.

Reviewer: Throughout the entire list, apply the following guidelines (make sure the dialog name is consistent with the list at the beginning of the page):

Defines the `Style` for the DIALOG NAME.

For example, "Defines the `Style` for the Edit Appointment Dialog."

Author: agree

* `EditRecurrenceDialogStyle`&mdash;Defines the Style that will be applied on the edit recurrence dialog control.
* `EditRecurrenceChoiceDialogStyle`&mdash;Defines the Style that will be applied on the edit recurrence choice dialog control.
* `DeleteAppointmentChoiceDialogStyle`&mdash;Defines the Style that will be applied on the delete appointment choice dialog control.
* `DeleteRecurrenceChoiceDialogStyle`&mdash;Defines the Style that will be applied on the delete recurrence choice dialog control.

> * `PreviewAppointmentDialogStyle`&mdash;Defines the Style that will be applied on the preview appointment dialog control. This dialog is only used in Android and iOS. 

Reviewer: Note that I'm using the name from the start of the page. Change as needed.

```suggestion
* `PreviewAppointmentDialogStyle`&mdash;(Android- and iOS-only)Defines the `Style` for the PreviewAppointment Dialog. 
```
